google/cloud/pubsub/subscriber_connection.h

```diff
@@ -74,6 +79,23 @@ std::shared_ptr<pubsub::SubscriberConnection> MakeSubscriberConnection(
     std::shared_ptr<SubscriberStub> stub,
     pubsub::ConnectionOptions const& options);
 
+template <typename Callable>
+std::unique_ptr<pubsub::SubscriberConnection::Callback> MakeSubscriberCallback(
+    Callable&& cb) {
+  class Wrapper : public pubsub::SubscriberConnection::Callback {
+   public:
+    explicit Wrapper(absl::decay_t<Callable> cb) : cb_(std::move(cb)) {}
```

Comment:
The argument of type `absl::decay_t<Callable>` is a value that will be move-constructed due to the one call site down on line 96. Then it will do a second move-construction into `cb_` in this c'tor initialization list. We could avoid the first move-construction by making this c'tor a template

```cc
...
  template <typename U>
  explicit Wrapper(U&& u) : cb_(std::forward<U>(u)) {}
...
```
Of course, doing this would probably require that this class be defined outside the function, becuase IIRC, we can't define templates within a function. But this is already in `pubsub_internal`, so maybe that's not a problem?

Anyway, WDYT?